File: app/src/main/java/com/starostinvlad/fan/Adapters/SearchListAdapter.java
package com.starostinvlad.fan.Adapters;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;
import com.starostinvlad.fan.BlurTransformation;
import com.starostinvlad.fan.GsonModels.News;
import com.starostinvlad.fan.R;

import java.util.List;


public class SearchListAdapter extends BaseAdapter {
    private final String TAG = getClass().getSimpleName();
    private List<News> list;

    public SearchListAdapter(List<News> list) {
        this.list = list;
    }

    @Override
    public int getCount() {
        return list.size();
    }

    @Override
    public Object getItem(int i) {
        return list.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        if (view == null) {
            view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.search_list_item, viewGroup, false);
        }

        TextView header = view.findViewById(R.id.search_item_header);
        header.setText(list.get(i).getTitle());

        TextView description = view.findViewById(R.id.search_item_description);
        description.setText(list.get(i).getSubTitle());

        ImageView imageView = view.findViewById(R.id.search_item_poster);
        Picasso.with(viewGroup.getContext())
                .load(list.get(i).getImage())
                .transform(new BlurTransformation(viewGroup.getContext()))
                .placeholder(R.color.cardview_dark_background)
                .into(imageView);

        return view;
    }
}

File: app/src/main/java/com/starostinvlad/fan/Adapters/SearchListAdapter.kt
package com.starostinvlad.fan.Adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import com.starostinvlad.fan.BlurTransformation
import com.starostinvlad.fan.GsonModels.News
import com.starostinvlad.fan.R


interface OnItemClickListener {
    fun onClick(item: News)
}

class SearchListAdapter(private val itemClickListener: OnItemClickListener) : RecyclerView.Adapter<SearchListAdapter.ViewHolder>() {
    private val TAG: String = this::class.simpleName!!
    var list: MutableList<News> = mutableListOf()
        set(value) {
            field.clear()
            field.addAll(value)
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.search_list_item, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        holder.apply {
            header.text = list[position].title
            description.text = list[position].subTitle
            Picasso.get()
                    .load(list[position].image)
                    .transform(BlurTransformation(holder.itemView.context))
                    .placeholder(R.color.cardview_dark_background)
                    .into(imageView)
            itemView.setOnClickListener {
                itemClickListener.onClick(list[position])
            }
        }


    }

    override fun getItemCount(): Int {
        return list.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val header = itemView.findViewById<TextView>(R.id.search_item_header)
        val description = itemView.findViewById<TextView>(R.id.search_item_description)
        val imageView = itemView.findViewById<ImageView>(R.id.search_item_poster)
    }
}